stratuslab/python/tar/slipstream_stratuslab/StratusLabIterClientCloud.py

```diff
@@ -184,21 +228,16 @@ def _stop_vms_by_ids(self, ids):
         """
         if not ids:
             return
-        configHolder = self.slConfigHolder.copy()
-        runner = self._get_stratuslab_runner(None, configHolder)
-        vm_infos = []
-        for vm_id in ids:
-            vm_info = runner.cloud._vmInfo(int(vm_id))
-            vm_infos.append(vm_info)
-        ids = map(int, ids)
-        for _id in ids[:]:
-            try:
-                runner.killInstances([_id])
-            except Exception as ex:
-                self._print_detail("Failed to terminate VM %s: %s" % (_id, str(ex)))
-                ids.remove(_id)
-        for vm_id in ids:
-            self._wait_vm_in_state(['Done', 'Failed'], runner, int(vm_id))
-            time.sleep(2)
-        for vm_info in vm_infos:
-            self._delete_volatile_disks(vm_info)
+        terminated_ids = super(StratusLabIterClientCloud, self)._stop_vms_by_ids(ids)
+        self._print_detail("Terminated VMs: %s" % terminated_ids)
+        volatile_disks_to_delete = self._get_volatile_disks_to_delete(terminated_ids)
+        if volatile_disks_to_delete:
+            # Need to wait until VMs are stopped.
+            runner = self._get_stratuslab_runner(None, self.slConfigHolder.copy())
+            for vm_id in terminated_ids:
+                self._wait_vm_in_state(['Done', 'Failed'], runner, int(vm_id))
```

Comment:
When deleted, VMs in OpenNebula enter final Done state.  I think this is covered here.